.github/workflows/test-llk-metal-integration.yaml

```
@@ -0,0 +1,272 @@
+name: Test LLK to Metal integration
+
+on:
+  workflow_dispatch:
+    inputs:
+      mirrored_branch:
+        description: 'Mirrored branch name (e.g., mirror/branch-name)'
+        required: true
+        type: string
+      run_all_post_commit:
+        description: 'Run all post-commit tests (Wormhole)'
+        required: false
+        type: boolean
+        default: false
+      run_blackhole_post_commit:
+        description: 'Run Blackhole post-commit tests'
+        required: false
+        type: boolean
+        default: false
+      workflow_timeout:
+        description: 'Timeout for workflows in minutes'
+        required: false
+        type: number
+        default: 240
+  workflow_call:
+    inputs:
+      mirrored_branch:
+        description: 'Mirrored branch name (e.g., mirror/branch-name)'
+        required: true
+        type: string
+      run_all_post_commit:
+        description: 'Run all post-commit tests (Wormhole)'
+        required: false
+        type: boolean
+        default: false
+      run_blackhole_post_commit:
+        description: 'Run Blackhole post-commit tests'
+        required: false
+        type: boolean
+        default: false
+      workflow_timeout:
+        description: 'Timeout for workflows in minutes'
+        required: false
+        type: number
+        default: 240
+
+env:
+  PARENT_BRANCH_NAME: test-llk-${{ inputs.mirrored_branch }}-${{ github.run_id }}
+  SUBMODULE_PATH: tt_metal/third_party/tt_llk
+  WORKFLOW_TIMEOUT: ${{ inputs.workflow_timeout || 240 }}
+  MAX_RETRIES: 3
+  POLL_INTERVAL: 120
+
+permissions:
+  contents: write
+  pull-requests: write
+  actions: write
+  issues: write
+  checks: read
+
+jobs:
+  setup-and-test:
+    runs-on: ubuntu-latest
+    outputs:
+      branch-exists: ${{ steps.check-branch.outputs.branch-exists }}
+      test-branch-name: ${{ steps.setup-branch.outputs.test-branch-name }}
+      parent-branch-name: ${{ env.PARENT_BRANCH_NAME }}
+    steps:
+      - name: Setup
+        uses: actions/checkout@v4
+        with:
+          repository: tenstorrent/tt-metal
+          submodules: recursive
+          token: ${{ secrets.TEMP_METAL_PAT }}
+          fetch-depth: 0
+          ref: main
+          clean: true
+      - name: Configure git
+        run: |
+          git config --global user.name "github-actions[bot]"
```

How about this for improved clarity?

```suggestion
          git config --global user.name "LLK Integration Tester [bot]"
```